src/main/java/org/team1540/base/power/PowerManager.java

```diff
@@ -52,18 +65,10 @@ public void run() {
     while (true) {
       // No whiles in here as that'd stop the last block from executing
       if (running) {
-        if (isSpiking()) {
-          if (theTimer.get() <= 0) {
-            // Calling the timer when it's already started seems to reset it.
-            theTimer.start();
-          }
-          if (timeHasPassed()) {
-            scalePower();
-          }
-        } else {
+        // ***EWWWW*** bad coding
+        if (!(checkPowerStatus(currentIsSpiking(), currentTimeHasPassed(), currentTimer) &&
```

Comment:
Literally no. This is totally not understandable. Doesn't matter if it doesn't look quite as nice, doing your management logic in an if statement check is just... no. Maybe the existence of `checkPowerStatus()` is okay, but not the way it's used now.

Reply:
I'm not saying it's not horrific, but doing an if-else if-else ifneither is terrible too–especially since that then requires maintaining a boolean flag and checking against it's current value. If you want changes here, help figure out a better solution.

Reply:
What's wrong with an if, else if, else chain exactly? No, it's not exactly the most incredibly elegant code, but just because you have to declare a variable does not mean that it's ugly.

Reply:
You have to check against the current state of the variable each time you run the next bit of the chain to see if you update it or not, which sucks.

Reply:
Could you maybe write something showing what that would look like? Don't push it, just paste it here. I'm having a hard time picturing what you're saying.

Reply:
Actually wait here's a cleaner way to do it with a boolean flag; just have it default true and set false.

```
boolean stopScaling = true;
if (isCurrentSpiking()) {
      // do stuff
      stopScaling = false;
    } else {
      // do stuff
    }

    if (isVoltageDipping()) {
      // do stuff
      stopScaling = false;
    } else {
      // do stuff
    }
        
    if (stopScaling) {
      stopScaling();
    }
}
```

We're doing that.

Reply:
Although this still has the issue of duplicated code since the boolean is being evaluated outside of the method, rather than just passing in the timer and booleans and letting it do that... Gotta figure that out.

Reply:
one thing that may, in fact, be a little bit more elegant is storing the two spike/dip values in variables and just setting `stopScaling = spiking || dipping` though that's not a pressing issue and is your call.

Reply:
IMO seems less elegant, but eh, they're both fine. I'm more concerned about the code duplication here.